slsim/lens_pop.py

```diff
@@ -284,8 +284,11 @@ def __init__(
                     sky_area=sky_area,
                 )
                 if source_type == "supernovae":
+                    redshift_sample = supernovae_catalog_class.host_galaxy_catalog()[
```

Comment:
will do this do certain calculations twice? Just checking. Perhaps we should also change the source_type for it as this particular type uses a host catalogue, if I am not misunderstood

Reply:
This does not do any calculation twice. Now, redshift is one of the inputs in supernovae_catalog() function in SupernovaeCatalog class. So, I just get redshift of host galaxies and supplied that as an input but one can supply any list of redshift here. We can replace this line by @cmistick 's redshift sample. This will not use any host galaxy information.